### PR TITLE
Support for refined library

### PR DESCRIPTION
**Pull request by danslapman**
This PR adds support for [refined](https://github.com/fthomas/refined) types

### CODE REVIEW

**Review by desavitsky — COMMENTED** [review]
Maybe add some tests?

**Comment by desavitsky** [comment]
I think that adding support for [iron](https://github.com/Iltotore/iron) might also be great

File: oolong-bson-refined/src/test/scala/oolong/bson/refined/RoundRobinSpec.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import eu.timepit.refined.numeric.*
+import oolong.bson.{*, given}
+import org.scalatest.TryValues
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import cats.scalatest.EitherValues
+
+class RoundRobinSpec extends AnyFunSuite with Matchers with TryValues with EitherValues  {
+  test("Refined serialization") {
+    val refinedVal: Int Refined Positive = refineV[Positive](5).value
```
Comment (desavitsky) [inline oolong-bson-refined/src/test/scala/oolong/bson/refined/RoundRobinSpec.scala:14]:
```scala
val refinedVal: Int Refined Positive = 5
```
Can't we write like that?

Comment (danslapman) [reply oolong-bson-refined/src/test/scala/oolong/bson/refined/RoundRobinSpec.scala:14]: This refined's macro is not yet re-written for Scala 3

**Pull request merged**